src/ConfigContext.ts

```diff
@@ -87,7 +91,12 @@ export function updateConfigContext(newContext: Partial<ConfigContext>): void {
     return;
   }
 
-  if (!validateEndpoint(newContext.ARM_ENDPOINT, allowedArmEndpoints)) {
+  if (
+    !validateEndpoint(
+      newContext.ARM_ENDPOINT,
+      configContext.allowedArmEndpoints ? configContext.allowedArmEndpoints : defaultAllowedArmEndpoints
```

Comment:
You can use the `||` operator for this line: `configContext.allowedArmEndpoints || defaultAllowedArmEndpoints`

Reply:
Updated.